### ISSUE TITLE
FT4.0 and FT2.0 accuracy contrast test

### ISSUE THREAD

**Issue by 520jefferson** [NVIDIA/FasterTransformer#215]
I use the transformer model, and the config as follows:
![image](https://user-images.githubusercontent.com/5691554/164583535-76fdb304-b03c-4161-b92f-b65509bd8913.png)

1, when using config2(filter_size = 4*hidden_size), i use FT2.0 to speed up the reasoning,  i compare the no original decoding (No optimized decoding) ​ results and the FT2.0 speed up  decoding results,  80% of the results are completely consistent, and it meet expectations, so i think the FT2.0 accuracy is okay with enough training.
2,when using config1, config2, config3, i compare the FT2.0 and FT4.0 proportion of the same decoding results  on the same testset. the config2 hidden size 48 is not supported by FT4.0 and we can ignore. In config1 and config3, we can see when the number of decoder layers is deepened, the inconsistency between FT2.0 and FT4.0  becomes worse even with enough training steps, and the FT4.0 results can not be accepted  after manual evaluation.
![image](https://user-images.githubusercontent.com/5691554/164584273-3c9c3410-d8df-4bac-a065-8e73ccad9906.png)
3, I also compare the  encoding output between FT2.0 and FT4.0, the diff is negligible, and i compare FT2.0 and FT4.0 results  on fp32  not with int8 and fp16, so according to my experiment, i find the decoding error  will expand when the decoder layer become bigger in FT4.0, if i want to use FT4.0 to optimize my config3 model, how should i resolve the accuracy problem?  and whether FT5.0 will resolve the accuracy problem when decoder layer increase?
![image](https://user-images.githubusercontent.com/5691554/164585467-5658a5ba-8781-4a62-a603-d03558dd061c.png)

**Comment by byshiue** [comment]
What about the accuracy on benchmark? FT doesn't guarantee that the results can be fully same. FT only provide same level accuracy on public testing data set. 
If you feel there are accuracy issue, please provide reproduce steps (docker, gpu, scripts, dataset) to reproduce your problem to show that there is obvious accuracy drop on v4.0 .

**Comment by 520jefferson** [comment]
I conduct the experiment on dialog generation task, we not only compare the proportion of consistency but also manual evaluation (judge whether the answer is   reasonable when given a input query), and the test set is randomly selected.

The environment is as follows:
docker: just using tensorflow1.14/5  noting special
gpu:v100
cuda:10.0
codes:opennmt
traning set:100w
test set(random select): 100 

The model structure and model change codes in google docs(op_encoderv4 and op_decodingv4)
https://docs.google.com/spreadsheets/d/1Pn5zC5MIHAwtrg4YSspK-QrT9oDfrhEgExWkZTKQEgg/edit?usp=sharing

And i will do more test when i am free , and i close the issue for now.
